Sources/SwiftProtobuf/TextDecoder.swift

```diff
@@ -21,67 +21,449 @@ import Swift
 /// pushback and convenience functions for iterating over complex
 /// structures.
 ///
-public struct TextDecoder: FieldDecoder {
-    private var scanner: TextScanner
-    public var complete: Bool {return scanner.complete}
-    public var rejectConflictingOneof: Bool {return true}
+public struct TextDecoder: Decoder {
+    internal var scanner: TextScanner
+    private var fieldCount = 0
+    private var terminator: UInt8?
+    private var fieldNameMap: FieldNameMap?
+    private var messageType: Message.Type?
```

Comment:
Text and JSON decoders now have common overall structure:  the "scanner" holds overall state (position within the stream) and the "decoder" holds the per-message state (field name map, field count, message type, etc).